src/_internal/clear/src/Clear.tsx

```diff
@@ -31,7 +31,13 @@ export default defineComponent({
   render () {
     const { clsPrefix } = this
     return (
-      <div class={`${clsPrefix}-base-clear`} data-clear>
+      <div
+        class={[
+          `${clsPrefix}-base-clear`,
+          this.show ? '' : `${clsPrefix}-base-clear--no-pointer`
+        ]}
```

Comment:
This is not the desired way, you may move `data-clear` to `.n-base-clear__clear`.

Since in date-picker we determine if clear is clicked by this.
![image](https://user-images.githubusercontent.com/18677354/123361205-37599200-d5a1-11eb-8574-c5e3e3f4a37f.png)